setup.py

```diff
@@ -5,12 +6,33 @@
 
 import Cython.Compiler.Options
 
-Cython.Compiler.Options.annotate = True
+Cython.Compiler.Options.annotate = False
 
-ext_modules = [Extension("pykrige.lib.cok", ["pykrige/lib/cok.pyx"], extra_compile_args=['-O3', '-march=native']),
-               Extension("pykrige.lib.lapack", ["pykrige/lib/lapack.pyx"], extra_compile_args=['-O2', '-march=native']),
-               Extension("pykrige.lib.variogram_models", ["pykrige/lib/variogram_models.pyx"],
-                         extra_compile_args=['-O2', '-march=native'])]
+if sys.platform != 'win32':
+    compile_args =  dict( extra_compile_args=['-O2', '-march=core2', '-mtune=corei7'],
+                 extra_link_args=['-O2', '-march=core2', '-mtune=corei7'])
+else:
+    compile_args = {}
+
+
+ext_modules = [Extension("pykrige.lib.cok",
+                         ["pykrige/lib/cok.pyx"],
+                                   **compile_args),
+               Extension("pykrige.lib.lapack",
+                         ["pykrige/lib/lapack.pyx"],
+                                   **compile_args),
+               Extension("pykrige.lib.variogram_models",
+                         ["pykrige/lib/variogram_models.pyx"],
+                                   **compile_args),]
+
+class build_ext_compiler_check(build_ext):
+    def build_extensions(self):
+        compiler = self.compiler

```

Comment:
Looking at attributes of `self.compiler`  might allow to determine whether a compiler is avalable...